.github/workflows/speech-test-data-ci.yml

```diff
@@ -141,90 +140,79 @@ jobs:
-            echo "::set-env name=MODEL_ID::$(echo $custom_speech_model_id)"
+            echo "MODEL_ID=$(echo $custom_speech_model_id)" >> $GITHUB_ENV
           else
             echo NO EXISTING CUSTOM SPEECH MODELS. TEST THE LATEST BASELINE MODEL.
-            echo "::set-env name=INITIAL_MODEL_EXISTS::false"
-            echo "::set-env name=IS_BASELINE_TEST::true"
+            echo "INITIAL_MODEL_EXISTS=false" >> $GITHUB_ENV
+            echo "IS_BASELINE_TEST=true" >> $GITHUB_ENV
           fi
-
       # If a benchmark model does not exist, or if a user pushed a tag beginning
       # with `BASELINE`, get the latest baseline model.
       #
       # Check that the baseline model has been successfully downloaded with
       # [[${my_variable//-/} =~ ^[[:xdigit:]]{32}$]] which will return true if
       # the variable is a valid GUID with 32 hexadecimal characters.
       - name: Get the baseline model
-        if: env.IS_BASELINE_TEST == 'true'
+        if: ${{ env.IS_BASELINE_TEST == 'true' }}
         run: |
-          speech model list-scenarios --locale ${{ env.SPEECH_LOCALE }} --simple > ${{ env.TEST_BUILD_FOLDER_PATH }}/baseline-models.txt
-          baseline_model_id=$(head -n 1 ${{ env.TEST_BUILD_FOLDER_PATH }}/baseline-models.txt)
+          spx csr model list --base --models --output json baseline_models.json
+          jq --arg LOC "$SPEECH_LOCALE" '.values[]|select(.locale==$LOC).self' baseline_models.json | tr -d \" > baseline_models.txt
+          rm baseline_models.json
+          tail -1 baseline_models.txt | tr -d '[:space:]' > my.test.model.url
+          baseline_model_id=$(cat my.test.model.url | tail -c 36)
           if ! [[ ${baseline_model_id//-/} =~ ^[[:xdigit:]]{32}$ ]]
           then
             echo "::error ::Failed to get the latest baseline model. Possibly re-run all jobs."
             exit 1
           fi
           echo TEST THE LATEST BASELINE MODEL WITH GUID: $baseline_model_id
-          echo "::set-env name=MODEL_ID::$(echo $baseline_model_id)"
-
+          echo "MODEL_ID=$(echo $baseline_model_id)" >> $GITHUB_ENV
       # Test with Speech.
       #
       # Check that the test has been successfully created with
       # [[${my_variable//-/} =~ ^[[:xdigit:]]{32}$]] which will return true if
       # the variable is a valid GUID with 32 hexadecimal characters.
       - name: Test the benchmark or baseline model
         run: |
-          speech test create -n test_from_${{ env.UNDERSCORE_EVENT_NAME }}_${{ env.EVENT_ID }} -a ${{ env.AUDIO_TRANS_TEST_ID }} -m ${{ env.MODEL_ID }} -lm ${{ env.MODEL_ID }} --wait > ${{ env.TEST_BUILD_FOLDER_PATH }}/test-output.txt
-          test_id=$(cat ${{ env.TEST_BUILD_FOLDER_PATH }}/test-output.txt | sed -n '3p')
+          spx csr evaluation create --name test_from_${UNDERSCORE_EVENT_NAME}_${EVENT_ID} --model1 @my.test.model.url --model2 @my.test.model.url --dataset @my.testing.datasets --output url @my.test.result --wait 
```

Comment:
What's happening with:

```bash
--model1 @my.test.model.url --model2 @my.test.model.url
```

It looks like it's testing a model against itself. I'm guessing the logic of that is fine, but it's probably confusing to look at two identical models in the Speech portal - or maybe it's a way to confirm you get a precise WER by testing twice.

Is it possible to delete `--model2 @my.test.model.url`?

Reply:
I've updated the code to correctly get a base model and use that url for the model1 parameter. From [the API reference](https://westus.dev.cognitive.microsoft.com/docs/services/speech-to-text-api-v3-0/operations/CreateEvaluation), model1 is intended to be a base model and model2, the custom model. The evaluation tests both.

Reply:
Ah awesome,, thanks!